### DISCUSSION TITLE
How to generate referential integrity DDL statements for CodeFirst scenario?

### DISCUSSION THREAD

**Discussion by akak1977** [dotnetcore/FreeSql#1878] [Q&A]
Hi!
Could anyone provide me an info, how to implement REFERENCES DDL statemens for postgre accordingly to navigation settings?
I have two classes (for example):
``` csharp
    public class Company 
    {
        public string Inn { get; set; }
        public string Name { get; set; }
    }

    public class Account
    {
        public string Number { get; set; }
        public double Amount { get; set; }
        public double AmountLocked { get; set; }
        public Company Company { get; set; }

        public Guid CompanyId { get; set; }
    }
```
I have the navigation set with fluent syntax:
``` csharp 
            fsql.CodeFirst.ConfigEntity<Company>(x =>
            {
                x.Property(p => p.Id).IsIdentity(true).IsPrimary(true);
            });


            fsql.CodeFirst.ConfigEntity<Account>(x =>
            {
                x.Property(p => p.Id).IsIdentity(true).IsPrimary(true);
                x.Navigate(p => p.Company, nameof(Account.CompanyId), nameof(Company.Id));
            });
```
Next, using _fsql.CodeFirst.GetComparisonDDLStatements_ I get:
``` SQL
CREATE TABLE IF NOT EXISTS "public"."account" (  
  "id" UUID, 
  "number" VARCHAR(255), 
  "amount" FLOAT8 NOT NULL, 
  "amountlocked" FLOAT8 NOT NULL, 
  "companyid" UUID NOT NULL, 
  CONSTRAINT "public_account_pkey" PRIMARY KEY ("id")
) WITH (OIDS=FALSE);

CREATE TABLE IF NOT EXISTS "public"."company" (  
  "id" UUID, 
  "inn" VARCHAR(255), 
  "name" VARCHAR(255), 
  CONSTRAINT "public_company_pkey" PRIMARY KEY ("id")
) WITH (OIDS=FALSE);
```

And the question: where is REFERENCES or specific FOREIGN KEY constraint for **account** table? What I'm doing wrong?
Thanks!

**Comment by 2881099** [comment]
The navigation properties are virtual relationships at the code layer, and the CodeFirst API does not create physical foreign keys for the database layer.

**Reply by akak1977** [reply]
Thanks for answer!
But, excuse me, why?
I can set up everything with FreeSQL ORM, including columns, types, indexes, etc. But I can't support ref integrity thru codefirst sync? Could you explain?

**Reply by 2881099** [reply]
This feature is expensive to implement, and there are too many databases to support. 

In addition, foreign keys do not affect the ORM and reduce write performance.
